lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+            total_word_counts_avg = 0
+        if len(self._formulas) != 0:
+            sums_avg = round((float(formulas_sum) / len(self._formulas)), 1)
+        else:
+            sums_avg = 0
+        self._averages.append("Averages")
+        for x in range(len(active_dicts)):
+            cat_count = sum([counter[x] for counter in self._counters])
+            if len(self._counters) != 0:
+                self._averages.append(round(
+                    float(cat_count) / len(self._counters), 1))
+            else:
+                self._averages.append(0)
+        self._averages.append(sums_avg)
+        self._averages.append(total_word_counts_avg)
+        self._averages.append(scores_avg)
+
+    def join_active_dicts(self) -> list:
+        """Joins all active dicts into on list of Phrase objects
+
+        :return: list of phrases contained in the active dictionaries
+        """
+        active_dicts = self.get_active_dicts()
+        dictionaries = [Phrase(content=phrase, dict_label=dictionary.label)
+                        for dictionary in active_dicts
+                        for phrase in dictionary.content]
+        dictionaries.sort(key=lambda x: len(x.content.split()), reverse=True)
+        return dictionaries
+
+    def to_html(self) -> str:
+        """Generates an html table from dataframe
+
+        :return: dataframe table in html
+        """
+        df = self.to_data_frame()
+        html = df.to_html(classes="table table-striped table-bordered",
+                          index=False)
+        return html
+
+    def to_data_frame(self) -> pd.DataFrame:
+        """Generates a dataframe containing all values stored in this class
+
+        :return: a data frame containing all values stored in this class
+        members
+        """
+        columns = ['Document Name'] + [dictionary.label for dictionary in
+                                       self.get_active_dicts()] + \
+                  ['formula', 'total word count', 'score']
+        indices = [file.label for file in self._corpus] + ['averages']
+        df = pd.DataFrame(columns=range(len(columns)),
+                          index=range(len(indices)))
+        # adds row to df with: file label, count for each dict,
+        # formula result, total word count, score
+        for i, (file, formula, score, counter) in enumerate(
+            zip(self._corpus, self._formulas,
+                self._scores, self._counters)):
+            df.xs(i)[0] = file.label
+            for j, (count) in enumerate(counter):
+                df.xs(i)[j + 1] = count
+            df.xs(i)[j + 2] = formula
+            df.xs(i)[j + 3] = file.total_word_counts
+            df.xs(i)[j + 4] = score
+        # add a row to df with the average of each column
+        for j, (average) in enumerate(self._averages):
+            df.xs(i + 1)[j] = average
+        df.columns = columns
+        return df
+
+    def is_secure(self) -> bool:
+        """Checks if the formula is secure
```

Comment:
Please define "secure" more explicitly. Looks like you mean something like "containing only names of uploaded dictionaries and approved math symbols"